aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -106,6 +117,9 @@ public Orchestrator(
             .merger(merger)
             .dataStoreConfigurationProvider(dataStoreConfigurationProvider)
             .build();
+        this.syncObserver = new SyncMetricsObserver(localStorageAdapter,
```

Comment:
It gets assigned here, but where is it used? Do you have behavior in the constructor? If so, can you move it to `start()` and `stop()` methods, and keep the constructor only for dependency wiring?

Other thought: it would be kinda nice if this could be `syncProcessor.addObserver(new SyncMtericsObserver())`, or if the `syncProcessor` just managed this thing internally, itself._

Reply:
I refactored the logic to make it a bit lighter in terms of dependencies. I created an accumulator class that keeps track of the sync metrics for an individual model. Rather than depending on the local storage adapter and creating an observer, I'm just passing in a reference to this new object into the merge method.